frame/system/extensions/check_genesis_test.go

```diff
@@ -123,6 +123,5 @@ func Test_CheckGenesis_Metadata(t *testing.T) {
 
 func setupCheckGenesis() CheckGenesis {
 	mockModule = new(mocks.SystemModule)
-
-	return NewCheckGenesis(mockModule)
+	return CheckGenesis{mockModule}
```

Comment:
Can we call the constructor and cast it to the type for coverage in all tests?
`return NewCheckGenesis(mockModule).(CheckGenesis)`

Reply:
I can do it in some of the tests, but for example here

```go
func setupCheckMortality() CheckMortality {
	mockModule = new(mocks.SystemModule)
	return CheckMortality{systemModule: mockModule}
}
```
I can't, because this method has pointer receiver `func (cm *CheckMortality) Decode`